### PR TITLE
Plugin Generator Updates

### PR DESCRIPTION
**Pull request by Fryguy**
Plugin Generator Updates

### CODE REVIEW

**Comment by Fryguy** [comment]
As I ran the plugin generator I adjusted the following:

- I moved some of the helper classes that were directly in the spec/spec_helper.rb into spec/support files
- I effectively undid #524, as I don't understand the rationale for why it was done, other than the original issue being from 2016, but much has changed in pluggability since then.  All of the other plugins follow the pattern as is done in this PR.
- I removed the coveralls bit.  We want to move away from coveralls towards codeclimate for coverage details.

**Comment by Fryguy** [comment]
@gtanzillo Please review.

File: lib/tasks/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+```ruby
+namespace "manageiq:content" do
+  desc "Explaining what the task does"
+  task :your_task do
+    # Task goes here
+  end
+end
+```
+
+Tasks places in the lib/tasks_private directory will be private to the plugin
```
Comment (gtanzillo) [inline lib/tasks/README.md:19]: This should read `Tasks placed ...`

Comment (Fryguy) [reply lib/tasks/README.md:19]: I'll have to change that in the generator and blast it out separately.  Let's leave it here for now.  I have a couple of other things to blast out in the next round.

**Pull request merged**